spirv_msl.cpp

```diff
@@ -8616,9 +8643,16 @@ void CompilerMSL::emit_instruction(const Instruction &instruction)
 		args.base.is_fetch = true;
 		args.coord = coord_id;
 		args.lod = lod;
+		if (needs_frag_discard_checks())
```

Comment:
This will break if emitted inside a continue block, control flow constructs is banned there.

Reply:
Alternatively, it might be possible to detect any complicated stores inside a continue block, and force the "complex loop" paths.

Reply:
I mostly did the former, but for `OpStore` I was forced to do the latter, due to the way `emit_instruction()` works.

Reply:
And it just occurred to me I forgot to add a test case for that. I meant to do that before updating...

Reply:
Added.